src/components/lava-coder/use-camera-controls.ts

```diff
@@ -1,21 +1,89 @@
-import { Cartesian3, Cartographic, CesiumWidget, Math as CSMath, sampleTerrainMostDetailed } from "@cesium/engine";
-import { useEffect } from "react";
+import {
+  Cartesian2, Cartesian3, Cartographic, CesiumWidget, Math as CSMath, HeadingPitchRange, sampleTerrainMostDetailed
+} from "@cesium/engine";
+import { useCallback, useEffect, useState } from "react";
+import { IOnDragArgs, useCesiumDragEvents } from "./use-cesium-drag-events";
+
+export const kDefaultCameraMode = "panning";
 
 const kDefaultXLng = -155.470;
 const kDefaultYLat = 19.150;
-const kMinDistanceAboveTerrain = 500;
-const kMaxDistanceAboveTerrain = 130000;
+const kMinDistanceAboveTerrain = 1000;
+const kMaxDistanceAboveTerrain = 135000;
+
+function getAngleFromCenter(pos: Cartesian2, center: Cartesian2) {
+  return Math.atan2(pos.y - center.y, pos.x - center.x);
+}
 
 export function useCameraControls(viewer: CesiumWidget | null, verticalExaggeration: number) {
 
-  function cameraMoveRate() {
+  const [cameraMode, setCameraMode] = useState<"pitch" | "heading" | "panning">(kDefaultCameraMode);
```

Comment:
I think you should make `"pitch" | "heading" | "panning"` a type since it's used in a couple of places.